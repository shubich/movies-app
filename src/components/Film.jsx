import React from 'react';
import { Link } from 'react-router-dom';

export const Film = ({id, title, year, genre, img}) => (
    <div className='film'>
        <Link to={'/film/'+id}>
            <img src={img} alt=""/>
        </Link>
        <div className='top-info'>
            <span className='title'>
                <Link to={'/film/'+id}>
                    {title}
                </Link>            
            </span>
            <span className='year'>{year}</span>
        </div>
        <div className='bottom-info'>
            <span className='genre'>{genre}</span>
        </div>
    </div>
);