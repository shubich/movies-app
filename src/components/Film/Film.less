.Film-Info {
    display: flex;
    flex-flow: row nowrap;
    align-items: baseline;
}

.Film-Title {
    flex: 1;
    font-size: 1.3em;
}

.Film-Year {
    border: 1px solid black;
    padding: 3px;
}