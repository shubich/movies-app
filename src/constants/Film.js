export const GET_FILM_DETAILS_ASYNC = 'GET_FILM_DETAILS_ASYNC';
export const GET_FILM_DETAILS_REQUEST = 'GET_FILM_DETAILS_REQUEST';
export const GET_FILM_DETAILS_SUCCESS = 'GET_FILM_DETAILS_SUCCESS';
export const GET_FILM_DETAILS_FAILURE = 'GET_FILM_DETAILS_FAILURE';
