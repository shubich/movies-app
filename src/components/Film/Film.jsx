import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import Poster from '../Poster';

import './Film.less';

const Film = ({
  film,
}) => (
  <div className="Film">
    <Link to={`/film/${film.id}`}>
      <Poster path={film.poster_path} />
    </Link>
    <div className="Film-Info">
      <Link className="Film-Title" to={`/film/${film.id}`}>
        {film.title}
      </Link>
      {
        film.release_date
        ? <span className="Film-Year">{film.release_date.split('-')[0]}</span>
        : null
      }
    </div>
  </div>
);

Film.propTypes = {
  film: PropTypes.shape({
    id: PropTypes.number,
    poster_path: PropTypes.string,
    title: PropTypes.string,
    release_date: PropTypes.string,
  }).isRequired,
};

export default Film;
