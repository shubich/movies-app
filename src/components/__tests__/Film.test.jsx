import React from 'react';
import { shallow } from 'enzyme';
import Film from '../Film';

let fullProps;
let partialProps;

describe('Film', () => {
  beforeEach(() => {
    fullProps = {
      film: {
        id: 1,
        title: 'Pulp Fiction',
        poster_path: '/img1234_0987',
        release_date: '1999',
      },
      onClick: jest.fn,
    };

    partialProps = {
      film: {
        id: 2,
        title: 'Taxi',
      },
      onClick: jest.fn,
    };
  });

  it('renders correctly with full props', () => {
    const wrapper = shallow(<Film {...fullProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with partial props', () => {
    const wrapper = shallow(<Film {...partialProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
