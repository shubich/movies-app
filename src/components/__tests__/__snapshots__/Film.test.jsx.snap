// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Film renders correctly with full props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Film
    film={
      Object {
        "id": 1,
        "poster_path": "/img1234_0987",
        "release_date": "1999",
        "title": "Pulp Fiction",
      }
    }
    onClick={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          replace={false}
          to="/film/1"
        >
          <Poster
            path="/img1234_0987"
          />
        </Link>,
        <div
          className="top-info"
        >
          <Link
            className="title"
            replace={false}
            to="/film/1"
          >
            Pulp Fiction
          </Link>
          <span
            className="year"
          >
            1999
          </span>
        </div>,
        <div
          className="bottom-info"
        />,
      ],
      "className": "film",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": <Poster
            path="/img1234_0987"
          />,
          "replace": false,
          "to": "/film/1",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "path": "/img1234_0987",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              className="title"
              replace={false}
              to="/film/1"
            >
              Pulp Fiction
            </Link>,
            <span
              className="year"
            >
              1999
            </span>,
          ],
          "className": "top-info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": "Pulp Fiction",
              "className": "title",
              "replace": false,
              "to": "/film/1",
            },
            "ref": null,
            "rendered": "Pulp Fiction",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "1999",
              "className": "year",
            },
            "ref": null,
            "rendered": "1999",
            "type": "span",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "className": "bottom-info",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            replace={false}
            to="/film/1"
          >
            <Poster
              path="/img1234_0987"
            />
          </Link>,
          <div
            className="top-info"
          >
            <Link
              className="title"
              replace={false}
              to="/film/1"
            >
              Pulp Fiction
            </Link>
            <span
              className="year"
            >
              1999
            </span>
          </div>,
          <div
            className="bottom-info"
          />,
        ],
        "className": "film",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": <Poster
              path="/img1234_0987"
            />,
            "replace": false,
            "to": "/film/1",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "path": "/img1234_0987",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                className="title"
                replace={false}
                to="/film/1"
              >
                Pulp Fiction
              </Link>,
              <span
                className="year"
              >
                1999
              </span>,
            ],
            "className": "top-info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": "Pulp Fiction",
                "className": "title",
                "replace": false,
                "to": "/film/1",
              },
              "ref": null,
              "rendered": "Pulp Fiction",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "1999",
                "className": "year",
              },
              "ref": null,
              "rendered": "1999",
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "className": "bottom-info",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Film renders correctly with partial props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Film
    film={
      Object {
        "id": 2,
        "title": "Taxi",
      }
    }
    onClick={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          replace={false}
          to="/film/2"
        >
          <Poster
            path=""
          />
        </Link>,
        <div
          className="top-info"
        >
          <Link
            className="title"
            replace={false}
            to="/film/2"
          >
            Taxi
          </Link>
        </div>,
        <div
          className="bottom-info"
        />,
      ],
      "className": "film",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": <Poster
            path=""
          />,
          "replace": false,
          "to": "/film/2",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "path": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              className="title"
              replace={false}
              to="/film/2"
            >
              Taxi
            </Link>,
            null,
          ],
          "className": "top-info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": "Taxi",
              "className": "title",
              "replace": false,
              "to": "/film/2",
            },
            "ref": null,
            "rendered": "Taxi",
            "type": [Function],
          },
          null,
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "className": "bottom-info",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            replace={false}
            to="/film/2"
          >
            <Poster
              path=""
            />
          </Link>,
          <div
            className="top-info"
          >
            <Link
              className="title"
              replace={false}
              to="/film/2"
            >
              Taxi
            </Link>
          </div>,
          <div
            className="bottom-info"
          />,
        ],
        "className": "film",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": <Poster
              path=""
            />,
            "replace": false,
            "to": "/film/2",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "path": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                className="title"
                replace={false}
                to="/film/2"
              >
                Taxi
              </Link>,
              null,
            ],
            "className": "top-info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": "Taxi",
                "className": "title",
                "replace": false,
                "to": "/film/2",
              },
              "ref": null,
              "rendered": "Taxi",
              "type": [Function],
            },
            null,
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "className": "bottom-info",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
